flink-end-to-end-tests/flink-sql-client-test/src/test/java/SqlClientITCase.java

```diff
@@ -290,10 +290,15 @@ private void verifyNumberOfResultRecords(String resultFilePath, int expectedNumb
         File tempOutputFile = new File(tempDir, "records.out");
         String tempOutputFilepath = tempOutputFile.toString();
         GenericContainer<?> taskManager = flink.getTaskManagers().get(0);
-        Thread.sleep(5000); // prevent NotFoundException: Status 404
-        taskManager.copyFileFromContainer(resultFilePath, tempOutputFilepath);
-
-        int numberOfResultRecords = UpsertTestFileUtil.getNumberOfRecords(tempOutputFile);
+        int numberOfResultRecords;
+        while (true) {
+            Thread.sleep(5000); // prevent NotFoundException: Status 404
```

Comment:
@zentol Any thoughts if/how we could get rid of the wait?

Reply:
cc @zentol

Reply:
At the end of the way you will need to poll _something_, be it for the output file or the status of the job.

But the sleep could be way lower; 50ms would be perfectly fine.

Reply:
@WencongLiu Could you lower the sleep?

Reply:
Done. thx @MartijnVisser @zentol